server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java

```diff
@@ -54,10 +56,20 @@
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
 import java.util.Collections;
+import java.util.List;
 import java.util.Map;
 
 public class DataSchemaTest
 {
+
+  private static final String VALID_DATASOURCE_CHARS_NAME = "alpha123..*~!@#&%^&*()-+ Россия\\ 한국 中国!";
+  private static final String VALID_DATASOURCE_CHARS_FOR_SERIALIZER = "alpha123..*~!@#&%^&*()-+ Россия\\\\ 한국 中国!";
```

Comment:
Suggest adding a helper method that escapes `VALID_DATASOURCE_CHARS_NAME`. The helper method can be used here as well as in `testSerializeWithInvalidDataSourceName()`

Reply:
👍